### DISCUSSION TITLE
How to get PortableExecutableReference from an Assembly?

### DISCUSSION THREAD

**Discussion by NMSAzulX** [dotnet/roslyn#54622] [Q&A]
Who can enumerate all the answers to this question? (I am optimizing my program that needs to get a reference from the assembly and save it.This is a bit difficult for me because the assembly cannot be converted into a stream.)

**Comment by ryzngard** [comment]
https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.metadatareference.createfromfile?view=roslyn-dotnet-3.9.0 

> This is a bit difficult for me because the assembly cannot be converted into a stream

The Assembly will still have to be read when using this API. Why can't you get a stream for it?

**Reply by NMSAzulX** [reply]
In some situations the "Location" of some assemblies is empty.
